blocks/hooks/custom-classname.js

```diff
@@ -0,0 +1,90 @@
+/**
+ * External dependencies
+ */
+import { assign } from 'lodash';
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { hasBlockSupport } from '../api';
+import InspectorControls from '../inspector-controls';
+import { extendElement } from './utils';
+
+/**
+ * Filters registered block settings, extending attributes with anchor using ID
+ * of the first node
+ *
+ * @param  {Object} settings Original block settings
+ * @return {Object}          Filtered block settings
+ */
+export function addAttribute( settings ) {
+	if ( hasBlockSupport( settings, 'customClassName', true ) ) {
+		// Use Lodash's assign to gracefully handle if attributes are undefined
+		settings.attributes = assign( settings.attributes, {
+			className: {
+				type: 'string',
+			},
+		} );
+	}
+
+	return settings;
+}
+
+/**
+ * Override the default edit UI to include a new block inspector control for
+ * assigning the anchor ID, if block supports anchor
+ *
+ * @param  {Element} element Original edit element
+ * @param  {Object}  props   Props passed to BlockEdit
+ * @return {Element}         Filtered edit element
+ */
+export function addInspectorControl( element, props ) {
+	if ( hasBlockSupport( props.name, 'customClassName', true ) && props.focus ) {
+		element = extendElement(
```

Comment:
I noticed that `extendElement` is breaking the cypress tests. You can just type "Enter" in a paragraph block to reproduce locally (JS error in the console). I suspect it's due to extra component "remounting" due to how we extend the base `BlockEdit`. Does this mean, it's not a great idea to add "filters" to render functions. Not sure how to fix this, ideas @aduth

Reply:
Seeing this as well. Also always focusing to the beginning of the text when clicking within. So far unsuccessful in debugging, but will continue investigating.

Reply:
I was able to fix this by avoiding updating the "keys". I was expecting the React "missing keys" warning but for some reason I'm not seeing it.

I'm not 100% confident here though. I would appreciate some help here.

Reply:
Actually, I was using the React's production build, so in "dev" I see the warnings. Problem is, adding the missing keys brings back the issue :(